File: src/main/java/org/pixeltime/enchantmentsenhance/event/enchantment/Lumberjack.java
package org.pixeltime.enchantmentsenhance.event.enchantment;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

import java.util.*;

public class Lumberjack implements Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    public void onBreak(final BlockBreakEvent blockBreakEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "lumberjack"));
        if (blockBreakEvent.isCancelled()) {
            return;
        }
        final Player player = blockBreakEvent.getPlayer();
        if (player.getItemInHand() != null && player.getItemInHand().hasItemMeta() && player.getItemInHand().getItemMeta().hasLore() && player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && blockBreakEvent.getBlock().getType() == Material.LOG) {
            final ArrayList<Material> list = new ArrayList<Material>();
            list.add(Material.LOG);
            list.add(Material.LEAVES);
            final Iterator<Block> iterator = this.getTree(blockBreakEvent.getBlock(), list).iterator();
            while (iterator.hasNext()) {
                iterator.next().breakNaturally();
            }
        }
    }

    private Set<Block> getNearbyBlocks(final Block block, final List<Material> list, final HashSet<Block> set) {
        for (int i = -1; i < 2; ++i) {
            for (int j = -1; j < 2; ++j) {
                for (int k = -1; k < 2; ++k) {
                    final Block block2 = block.getLocation().clone().add((double) i, (double) j, (double) k).getBlock();
                    if (block2 != null && !set.contains(block2) && list.contains(block2.getType())) {
                        set.add(block2);
                        set.addAll(this.getNearbyBlocks(block2, list, set));
                    }
                }
            }
        }
        return set;
    }

    public Set<Block> getTree(final Block block, final List<Material> list) {
        return this.getNearbyBlocks(block, list, new HashSet<Block>());
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Lumberjack.kt
package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager
import java.util.*

class Lumberjack : Listener {
    private val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "lumberjack"))
    @EventHandler(priority = EventPriority.MONITOR)
    fun onBreak(blockBreakEvent: BlockBreakEvent) {
        if (blockBreakEvent.isCancelled) {
            return
        }
        val player = blockBreakEvent.player


        try {
            val armorContents = player.inventory.armorContents + IM.getAccessorySlots(player)
            for (i in armorContents.indices) {
                val itemStack = armorContents[i]
                if (itemStack.hasItemMeta() && itemStack.itemMeta.hasLore()) {
                    val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
                    if (level > 0 && blockBreakEvent.block.type == Material.LOG) {
                        val list = ArrayList<Material>()
                        list.add(Material.LOG)
                        list.add(Material.LEAVES)
                        val iterator = this.getTree(blockBreakEvent.block, list).iterator()
                        while (iterator.hasNext()) {
                            iterator.next().breakNaturally()
                        }
                    }
                }
            }
        } catch (ex: Exception) {
        }
    }

    private fun getNearbyBlocks(block: Block, list: List<Material>, set: HashSet<Block>): Set<Block> {
        for (i in -1..1) {
            for (j in -1..1) {
                for (k in -1..1) {
                    val block2 = block.location.clone().add(i.toDouble(), j.toDouble(), k.toDouble()).block
                    if (block2 != null && !set.contains(block2) && list.contains(block2.type)) {
                        set.add(block2)
                        set.addAll(this.getNearbyBlocks(block2, list, set))
                    }
                }
            }
        }
        return set
    }

    fun getTree(block: Block, list: List<Material>): Set<Block> {
        return this.getNearbyBlocks(block, list, HashSet())
    }
}
